src/Relation/Binary/Construct/SetoidFromPartialSetoid.agda

```diff
@@ -0,0 +1,61 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Conversion of a PartialSetoid into a Setoid
+------------------------------------------------------------------------
+
+open import Relation.Binary.Bundles using (PartialSetoid; Setoid)
+
+module Relation.Binary.Construct.SetoidFromPartialSetoid
+  {a ℓ} (S : PartialSetoid a ℓ) where
+
+open import Data.Product.Base using (_,_; Σ; proj₁)
+open import Function.Base using (id)
+open import Relation.Binary.Core using (Rel)
+open import Relation.Binary.Definitions using (Reflexive)
+open import Relation.Binary.Structures
+  using (IsPartialEquivalence; IsEquivalence)
+open import Relation.Binary.Morphism.Structures
+  using (IsRelHomomorphism; IsRelMonomorphism)
+
+private
+  module S = PartialSetoid S
+
+  variable
+    x y z : S.Carrier
+
+------------------------------------------------------------------------
+-- Definitions
+
+Carrier : Set _
+Carrier = Σ S.Carrier λ x → x S.≈ x
+
+_≈_ : Rel Carrier _
+x≈x@(x , _) ≈ y≈y@(y , _) = x S.≈ y
```

Comment:
```
x ≈ y = proj₁ x S.≈ proj₁ y
```
?  Or even 
```
_≈_ = S._≈_ on proj₁
```
I certainly don't like binding names that are not used on the rhs as you do here.